### ISSUE TITLE
Add ability to pause between keystrokes arbitrarily anywhere

### ISSUE THREAD

**Issue by momomo** [pqrs-org/Karabiner-Elements#3807]
Take this example: 

```

{
        // Does not really work well
        "type": "basic",
        
        "from": {
            "key_code": "period",
            "modifiers": {
                "mandatory": [
                    "left_shift",
                    "left_option"
                ]
            }
        },
        
        "to": [
            
            {
                "key_code": "right_arrow",
                "modifiers": [
                    "left_option",
                    
                    "right_control",
                    "right_shift"
                ]
            },

            // Pause here for 0.25 seconds 
            
            {
                "key_code": "tab",
                "modifiers": [
                    "left_command"
                ]
            },
          
            // Pause here for 0.25 seconds 

            {
                "key_code": "left_arrow",
                "modifiers": [
                    "left_option",
                    
                    "right_control",
                    "right_shift"
                ]
            }
            
        ]
    }

```

The issue is that the keys are invoked way too fast after each other thus not allowing the computer to execute the intended actions fully. 

Ideally we would like to pause between certain keystrokes as a program might be running that is laggy or slow. 

It could be something along the lines 

```
{
   wait: {
        time: 500
   }
   
}
```

or 

```
{
   wait: {
        until: application is in front "abcdef" 
   }
   
}
```

you would likely have to rework how things are executed or add a whole new to_pausable logic to handle it to pull this off. 

However, the need for something like this has arisen multiple times so it is a very valid usecase. Sometimes, there needs to be a delay when one keystroke or shortcut is to trigger multiple keystrokes that do things that might take certain time. We the developer is already in a messy situation utilizing karabiner so its just about enabling the workflow to improve further. 

If you don't see the usecase, realize the use case is not about you.

**Comment by ToeKneeFan** [comment]
For now, you can emulate this with `to_delayed_action` ([link](https://karabiner-elements.pqrs.org/docs/json/complex-modifications-manipulator-definition/to-delayed-action/)). The delay can be configured with the `basic.to_delayed_action_delay_milliseconds` parameter. The documentation says "`to_delayed_action` posts events after 500 milliseconds from the `from` key is pressed", so I am not sure if it guarantees ordering (may be based on delay from `from` rather than last key), but well-chosen delays should essentially give you what you want.

**Comment by momomo** [comment]
@ToeKneeFan 

No, that only works for first keystroke. I am talking about pausing between subsequent keystrokes. 

If you type on a keyboard, you might type a ... wait for a menu, then enter next keystroke ... so a wait not just on the first, but even later ...

**Comment by ToeKneeFan** [comment]
Ah, good point. One suggestion [here](https://github.com/pqrs-org/Karabiner-Elements/issues/2664) is to just hold down a key longer with `hold_down_milliseconds` (not one where holding down the key would affect the output, like a letter, but instead, for example, a modifier key). But I can see the merits of your feature request to avoid some side effects of this workaround.

**Comment by hyuri** [comment]
Came here to suggest this. Very needed in some cases, like when you have a sequence of shortcuts and one of them triggers opening a dialog with animation — so the next shortcut in the chain isn't activated because the animation hasn't finished it so the dialog isn't ready.

**Comment by artemcrum** [comment]
Any news on this? One more vote for this enhancement.

**Comment by tekezo** [comment]
to.hold_down_milliseconds is the feature intended for this.
https://karabiner-elements.pqrs.org/docs/json/complex-modifications-manipulator-definition/to/hold-down-milliseconds/

**Comment by clementaudic** [comment]
Hi @tekezo,

The problem with `to.hold-down-milliseconds` is that the key is repeated while being held.

For instance, with the following configuration, the `return_or_enter` key will be repeated several times during the 500 ms hold: 
```
{
    "from": { "key_code": "a" },
    "to": [
        {
            "hold_down_milliseconds": 500,
            "key_code": "return_or_enter"
        },
        { "key_code": "a" }
    ],
    "type": "basic"
}
```
What if we want to wait for 500 ms but **do not** want the `return_or_enter` key to repeat?

This becomes even more problematic when the macOS option to allow letter keys to repeat while holding (instead of opening the accent pop-up) is enabled. (This option can be enabled with `defaults write -g ApplePressAndHoldEnabled -bool false` and requires a restart of your Mac)

Please consider reopening this issue.

**Comment by ToeKneeFan** [comment]
@clementaudic The suggestion is to hold down some other key that will do nothing, not the enter key. This effectively inserts a pause. Though I do think the feature request is a good idea to avoid side effects as well.

**Comment by clementaudic** [comment]
Thanks @ToeKneeFan, I didn't think to apply `to.hold_down_milliseconds` to a key that does nothing like `vk_none`.

The following configuration resolves the problem I mentioned in my previous comment:
```
{
    "from": { "key_code": "a" },
    "to": [
        { "key_code": "return_or_enter" },
        {
            "hold_down_milliseconds": 500,
            "key_code": "vk_none"
        },
        { "key_code": "a" }
    ],
    "type": "basic"
}
```
